main.ts

```diff
@@ -63,6 +64,7 @@ const DEFAULT_SETTINGS: S3UploaderSettings = {
 	useCustomEndpoint: false,
 	customEndpoint: "",
 	forcePathStyle: false,
+	customImageUrl: "",
```

Comment:
I prefer to use an explicit bool setting like with `customEndpoint` and `useCustomEndpoint` where we check whether the setting should be used, and only then invoke the required logic.  As it is in this PR, the use of the customUrl is implicit based on whether the value is "" or not.  This relies on the falsyness of "", which is not explicit, and inconsistent with the existing implementation.